### ISSUE TITLE
Unable to create Isaacs' comma-first style

### ISSUE THREAD

**Issue by morinted** [eslint/eslint#9380]
* **ESLint Version:** 4.8.0
* **Node Version:** 8.3.0
* **npm Version:** 5.4.1

**What parser (default,  etc.) are you using?**

Babel-ESLint

<details>
<summary>Configuration</summary>

```js
{ "parser": "babel-eslint"
, "extends": "eslint:recommended"
, "rules":
  {
  , "indent":
    [ "error"
    , 2
    , { "SwitchCase": 1
      , "ObjectExpression": 1
      }
    ]
  , "no-unexpected-multiline": 1
  , "no-multi-spaces": 1
  , "space-in-parens": 1
  , "no-trailing-spaces": 2
  , "comma-style":
    [ 2
    , "first"
    ]
  , "comma-spacing": 2
, "env":
  { "node": false
  , "browser": true
  , "es6": true
  }
, "plugins":
  [ "eslint-plugin-react"
  , "eslint-plugin-babel"
  ]
}
```

</details>

**Code:**

```js
function test() {
  return (
    { test: 'thing'
    , result: 'other thing'
    , sampleArray:
      [ 0
      , 1
      , 2
      ]
    }
  )
}
```

**What did you expect to happen?**

No warnings or errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
   93:1   error    Expected indentation of 6 spaces but found 4                                   indent
   94:1   error    Expected indentation of 6 spaces but found 4                                   indent
   96:1   error    Expected indentation of 8 spaces but found 6                                   indent
   97:1   error    Expected indentation of 8 spaces but found 6                                   indent
```

------------------

Seeing as the [Isaacs style](https://gist.github.com/isaacs/357981#file-comma-first-var-js-L175) is linked from the [comma-style rule](https://eslint.org/docs/rules/comma-style#further-reading), I think it would make sense to support it as it was intended. The recommended "fix" of:

```js
function test() {
  return (
    { test: 'thing'
      , result: 'other thing'
      , sampleArray:
      [ 0
        , 1
        , 2
      ]
    }
  )
}
```

makes no sense to me.

I'm really hoping this is just some kind of configuration issue that I can work around with rules. Any help is appreciated!

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended, but you can prevent errors from being reported by disabling the `indent` rule (with `indent: "off"`  in your config file).

**Comment by morinted** [comment]
So there is no way to configure the behavior of indent to support this?

If this is my ultimate goal, is there any way I could work the behavior
into a plugin (I haven't studied the eslint plugin API, so I'm just looking
for a binary yes/no as to whether it is possible).

**Comment by not-an-aardvark** [comment]
Actually, I think I may have been mistaken; you can enforce this indentation with the `[error, 2, {ObjectExpression: 0, ArrayExpression: 0}]` configuration for the `indent` rule.

**Comment by morinted** [comment]
Okay, that seems to be closer to convenient.

In order to stop next objects and arrays from an issue, I had to add some ignoredNodes rules, but otherwise this seems workable and more strict than 3.x. Thanks for your help!
